storage/bucket.go

```diff
@@ -455,6 +455,10 @@ type LifecycleCondition struct {
 	// the specified date in UTC.
 	CreatedBefore time.Time
 
+	// DaysSinceNoncurrentTime is the days elapsed since the noncurrent timestamp
```

Comment:
Is "noncurrent" a well known term for objects? It seems like it is, but I wanted to confirm because I haven't seen it before.

Reply:
It's new and will be documented along with the feature support in Cloud Storage docs. 

@jkwlui will have more context.

It represents "old versions" when object versioning is enabled for buckets.

Reply:
The concept of "noncurrent" version of an object already exists - https://cloud.google.com/storage/docs/object-versioning#intro. My opinion here is that users who wish to add this conditional would already be familiar with object versioning and the effect of this flag.